symengine/basic_to_upoly.cpp

```diff
@@ -0,0 +1,363 @@
+#include <symengine/visitor.h>
+
+namespace SymEngine
+{
+
+// returns the integer if `x` is a positive integer else
+// returns `-1` if negative integer, `0` if not an integer
+inline int positive_integer(const RCP<const Basic> &x)
+{
+    if (is_a<const Integer>(*x)) {
+        int tmp = rcp_static_cast<const Integer>(x)->as_int();
+        if (tmp > 0)
+            return tmp;
+        else
+            return -1;
+    }
+    return 0;
+}
+
+class UIntPolyGenaratorVisitor : public BaseVisitor<UIntPolyGenaratorVisitor>
+{
+private:
+    RCP<const Basic> res;
+    bool inpow;
+
+public:
+    RCP<const Basic> apply(const Basic &b, bool inpow_)
+    {
+        inpow = inpow_;
+        b.accept(*this);
+        return res;
+    }
+
+    void bvisit(const Pow &x)
+    {
+        if (is_a<const Rational>(*x.get_base())
+            or is_a<const Integer>(*x.get_base())) {
+            res = pow(x.get_base(), _find_gen_uintpoly(x.get_exp(), true));
+        } else if (is_a<const Rational>(*x.get_exp())
+                   or is_a<const Integer>(*x.get_exp())) {
+
+            int tmp = positive_integer(x.get_exp());
+            if (tmp != 0) {
+                if (tmp > 0)
+                    res = _find_gen_uintpoly(x.get_base());
+                else
+                    res = x.rcp_from_this();
+                return;
+            }
```

Comment:
You are right. In Line 47, you can do `div(one, x.get_base())` instead of `x.rcp_from_this()`